File: axvoice.android/src/com/axia/VoiceItem.java
package com.axia;

import java.io.File;

public class VoiceItem {
	public enum LocalStatus { NO_LOCAL_FILE, RECORDING, STOP_RECORDING, DOWNLOADING, HAS_LOCAL_FILE }
	public enum ServerStatus { NO_SERVER_FILE, UPLOADING, HAS_SERVER_FILE }
	public enum TextStatus { NO_TEXT, CONVERTING, HAS_TEXT }
	
	private long voiceID;				// unique id (local)
	private String localFilePath;		// local file path
	private String fileMD5;				// hash file 
	private String serverURL;			// server URL
	private String text;				// text
	private LocalStatus localStatus;	// local file status
	private ServerStatus serverStatus;	// server file status
	private TextStatus textStatus;		// text status
	public AudioRecorder recorder;		// recorder object
	
	public long getVoiceID() { return voiceID; }
	public String getLocalFile() { return localFilePath; }
	public String getServerFile() { return serverURL; }
	public String getMD5() { return fileMD5; }
	public String getText() { return text; }
	public LocalStatus getLocalStatus() { return localStatus; }
	public ServerStatus getServerStatus() { return serverStatus; }
	public TextStatus getTextStatus() { return textStatus; }
	
	public void startRecord() {
		this.localStatus = LocalStatus.RECORDING;
	}
	
	public void stopRecord() {
		this.localStatus = LocalStatus.STOP_RECORDING;
	}
	
	public void confirmLocalFile(String md5)
	{ 
		this.fileMD5 = md5; 
		this.localStatus = LocalStatus.HAS_LOCAL_FILE;
	}
	
	public void beginUpload() {
		this.serverStatus = ServerStatus.UPLOADING;
	}
	
	public void updateServerURL(String serverURL) {
		this.serverURL = serverURL;
		this.serverStatus = ServerStatus.HAS_SERVER_FILE;
	}
	
	public void beginDownload() {
		this.localStatus = LocalStatus.DOWNLOADING;
	}
	
	public void beginConvertToText() {
		//TODO: check Status
		this.textStatus = TextStatus.CONVERTING;
	}
	
	public void setText(boolean success, String text) {
		if(success) {
			this.textStatus = TextStatus.HAS_TEXT;
			this.text = text;
		} else {
			this.textStatus = TextStatus.NO_TEXT;
		}
	}
	
	//Create new item
	public VoiceItem(long voiceID, 
			String localFilePath, 
			String fileMD5, 
			String serverURL,
			String text,
			LocalStatus localStatus, 
			ServerStatus serverStatus,
			TextStatus textStatus)
	{
		this.voiceID = voiceID;
		this.localFilePath = localFilePath;
		this.fileMD5 = fileMD5;
		this.serverURL = serverURL;
		this.text = text;
		this.localStatus = localStatus;
		this.serverStatus = serverStatus;
		this.textStatus = textStatus;
		
		this.recorder = null;
	}
}

File: axvoice.ios/axvoice.ios/VoiceItem.m
//
//  VoiceItem.m
//  axvoice.ios
//
//  Created by JinChao on 14/12/24.
//  Copyright (c) 2014年 thecodeway. All rights reserved.
//

#import <Foundation/Foundation.h>
#import "VoiceItem.h"

@interface VoiceItem()
{
    uint _voiceID;
    NSString *_localWavFile;
    NSString *_localAmrFile;
    NSString *_fileMD5;
    NSString *_serverURL;
    NSString *_text;
    LocalStatus _localStatus;
    ServerStatus _serverStatus;
    ToTextStatus _toTextStatus;
}


@end

@implementation VoiceItem
@synthesize voiceID = _voiceID;
@synthesize localWavFile = _localWavFile;
@synthesize localAmrFile = _localAmrFile;
@synthesize serverURL = _serverURL;
@synthesize fileMD5 = _fileMD5;
@synthesize text = _text;
@synthesize localStatus = _localStatus;
@synthesize serverStatus = _serverStatus;
@synthesize toTextStatus = _toTextStatus;

- (id)initWithParam : (uint)voiceID
       localWavFile : (NSString*) localWavFile
        localAmrFile: (NSString*)localAmrFile
          serverURL : (NSString*) serverURL
            fileMD5 : (NSString*) fileMD5
               text : (NSString*) text
        localStatus : (LocalStatus) localStatus
       serverStatus : (ServerStatus) serverStatus
       toTextStatus : (ToTextStatus)toTextStatus
{
    if (self = [super init]) {
        
        _voiceID = voiceID;
        _localWavFile = localWavFile;
        _localAmrFile = localAmrFile;
        _serverURL = serverURL;
        _fileMD5 = fileMD5;
        _text = text;
        _localStatus = localStatus;
        _serverStatus = serverStatus;
        _toTextStatus = toTextStatus;
    }
    return self;
}

- (void)startRecord
{
    //TODO: Check status
    _localStatus = RECORDING;
}

- (void)stopRecord
{
    //TODO: Check status
    _localStatus = STOP_RECORDING;
}

- (void)confirmLocalFile :  (NSString*) amrMD5
{
    //TODO: Check status
    _fileMD5 = amrMD5;
    _localStatus = HAS_LOCAL_FILE;
}

- (void)beginUpload
{
    if([self localStatus] != HAS_LOCAL_FILE ||
       [self serverStatus] != NO_SERVER_FILE)
    {
        return; //TODO: ERROR
    }
    
    _serverStatus = UPLOADING;
}

- (void)updateServerURL:(NSString *)serverURL
{
    //TODO: Check status
    _serverURL = serverURL;
    _serverStatus = HAS_SERVER_FILE;
}

- (void)beginDownload
{
    //TODO: Check status
    _localStatus = DOWNLOADING;
}

- (void)beginConvertToText
{
    if([self localStatus] != HAS_LOCAL_FILE ||
       [self toTextStatus] != NO_TEXT)
    {
        return; //TODO: ERROR
    }
    _toTextStatus = CONVERTING;
}

- (void)setText : (bool) success
           text : (NSString *)text
{
    if([self localStatus] != HAS_LOCAL_FILE ||
       [self toTextStatus] != CONVERTING)
    {
        return; //TODO: ERROR
    }
    
    if(success) {
        _toTextStatus = HAS_TEXT;
        _text = text;
    }else {
        _toTextStatus = NO_TEXT;
    }
}
@end